src/lib/common/include/sol-log.h

```diff
@@ -468,14 +468,55 @@ sol_log_domain_init_level(struct sol_log_domain *domain)
  *
  * @see SOL_CRI(), SOL_ERR(), SOL_WRN(), SOL_INF() and SOL_DBG().
  */
+#ifdef SOL_LOG_FILES
+#ifdef SOL_LOG_FUNCTIONS
 #define SOL_LOG(level, fmt, ...) \
     do { \
         if (SOL_LOG_LEVEL_POSSIBLE(level)) { \
             sol_log_print(SOL_LOG_DOMAIN, level, \
-                __FILE__, __PRETTY_FUNCTION__, __LINE__, \
+                __FILE__, \
+                __PRETTY_FUNCTION__, \
+                __LINE__, \
```

Comment:
Would changing __FILE__ and friends to something like LOG_FILE that is defined to __FILE__ or NULL accordingly work?